integration/fuse/src/main/java/alluxio/fuse/file/FuseFileInOrOutStream.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.fuse.file;
+
+import alluxio.AlluxioURI;
+import alluxio.client.file.FileSystem;
+import alluxio.client.file.URIStatus;
+import alluxio.exception.AlluxioException;
+import alluxio.fuse.AlluxioFuseOpenUtils;
+import alluxio.fuse.auth.AuthPolicy;
+
+import jnr.constants.platform.OpenFlags;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * An implementation for {@link FuseFileStream} for read only or write only workloads.
+ * Fuse can open file for reading and writing concurrently but Alluxio only support
+ * read-only or write-only workloads. This class will treat the stream as read-only or write-only
+ * workloads based on whether truncate is called and the first operation is read or write.
+ */
+@ThreadSafe
+public class FuseFileInOrOutStream implements FuseFileStream {
+  private static final Logger LOG = LoggerFactory.getLogger(FuseFileInOrOutStream.class);
+  private final FileSystem mFileSystem;
+  private final AlluxioURI mUri;
+  private final long mMode;
+  private final AuthPolicy mAuthPolicy;
+
+  // TODO(lu) reconsider whether to store the status.
+  // If store status, then we can directly provide to all getattr
+  private final URIStatus mStatus;
```

Comment:
// TODO(lu) reconsider whether to store the status.
  // If store status, then we can directly provide to all getattr